Use ActionView::TemplateDetails for handling format and variant (#2156)
ViewComponent/view_component: docs/CHANGELOG.md

    *Joel Hawksley*

> * BREAKING: Remove support for variant names containing `.`
> 
>   This feature isn't supported by Rails and prevents the use of Rails logic
>   for file name parsing.

Reviewer: ```suggestion
* BREAKING: Remove support for variant names containing `.`, which isn't supported by Rails and prevents the use of Rails logic for file name parsing.
```

>   *Stephen Nelson*

Reviewer: ```suggestion
    *Stephen Nelson*
```
